public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -1215,3 +1206,24 @@ function getLogLevelFromLabels(labels: Labels): LogLevel {
   }
   return levelLabel ? getLogLevelFromKey(labels[levelLabel]) : LogLevel.unknown;
 }
+
+function getLabelTypeFromFrame(labelKey: string, frame?: DataFrame, index?: number): null | LabelType {
```

Comment:
Maybe move to utils? (I hate the `utils` concept but I would put it elsewhere).

Reply:
Moved it to `languageUtils` in 95f05a6face0ba435c7c74366b962561ae168c59